Automattic/jetpack
# Integrate jetpack-ip package

This PR introduces the new `jetpack-ip` package in places where packages and plugins are currently relying on various shared functions from the brute force protection module.

## Proposed changes:

* Adds the `jetpack-ip` dependency to `packages/status`, `packages/sync`, `packages/waf`, and `plugins/jetpack`.
* Replaces all uses of the shared brute force protection functions using the `Automattic\Jetpack\IP\Utils` class methods.
* Remove all of the now-unused IP-related code from the brute force protection's `shared-functions.php`.
* Updates `composer.lock` files for plugins implementing the affected packages.

### Other information:


## Jetpack product discussion



- peb6dq-EJ-p2

## Does this pull request change what data or activity we track or use?



- No

## Testing instructions:






* Start up a Jurassic Ninja instance using the beta plugin on this Jetpack branch
* Enable Jetpack and connect your user
* Verify that Brute force protection is enabled and is functioning as expected
  * Modify and save settings (i.e. toggle module on and off, update IP allow list)
  * Ensure the math auth displays after several failed login attempts
  * Ensure the hard block notice displays after an appropriate number of failed attempts and that the recovery process succeeds

**WPCOM/Calypso Testing Instructions**
* Apply `bin/jetpack-downloader test jetpack add/integrate-ip-package` to your WPCOM sandbox.
* Sandbox `public-api.wordpress.com`
* Check out `https://wordpress.com/settings/security/[YOUR_SITE_URL]` and validate the Brute Force Protection settings function as expected.
